code/modules/shuttle/shuttle.dm

```diff
@@ -0,0 +1,697 @@
+		return 1
+	if(istype(S, /obj/docking_port/stationary/transit))
+		return 0
+	//check dock is big enough to contain us
+	if(dwidth > S.dwidth)
+		return 2
+	if(width-dwidth > S.width-S.dwidth)
+		return 3
+	if(dheight > S.dheight)
+		return 4
+	if(height-dheight > S.height-S.dheight)
+		return 5
+	//check the dock isn't occupied
+	if(S.get_docked())
+		return 6
+	return 0	//0 means we can dock
+
+//call the shuttle to destination S
+/obj/docking_port/mobile/proc/request(obj/docking_port/stationary/S)
+	if(canDock(S))
+		. = 1
+		throw EXCEPTION("request(): shuttle cannot dock")
+		return 1	//we can't dock at S
+
+	switch(mode)
+		if(SHUTTLE_CALL)
+			if(S == destination)
+				if(world.time <= timer)
+					timer = world.time
+			else
+				destination = S
+				timer = world.time
+		if(SHUTTLE_RECALL)
+			if(S == destination)
+				timer = world.time - timeLeft(1)
+			else
+				destination = S
+				timer = world.time
+			mode = SHUTTLE_CALL
+		else
+			destination = S
+			mode = SHUTTLE_CALL
+			timer = world.time
+			enterTransit()		//hyperspace
+
+//recall the shuttle to where it was previously
+/obj/docking_port/mobile/proc/cancel()
+	if(mode != SHUTTLE_CALL)
+		return
+
+	timer = world.time - timeLeft(1)
+	mode = SHUTTLE_RECALL
+
+/obj/docking_port/mobile/proc/enterTransit()
+	previous = null
+//		if(!destination)
+//			return
+	var/obj/docking_port/stationary/S0 = get_docked()
+	var/obj/docking_port/stationary/S1 = findTransitDock()
+	if(S1)
+		if(dock(S1))
+			WARNING("shuttle \"[id]\" could not enter transit space. Docked at [S0 ? S0.id : "null"]. Transit dock [S1 ? S1.id : "null"].")
+		else
+			previous = S0
+	else
+		WARNING("shuttle \"[id]\" could not enter transit space. S0=[S0 ? S0.id : "null"] S1=[S1 ? S1.id : "null"]")
+
+//default shuttleRotate
+/atom/proc/shuttleRotate(rotation)
+	//rotate our direction
+	dir = angle2dir(rotation+dir2angle(dir))
+
+	//rotate the pixel offsets too.
+	if (pixel_x || pixel_y)
+		if (rotation < 0)
+			rotation += 360
+		for (var/turntimes=rotation/90;turntimes>0;turntimes--)
+			var/oldPX = pixel_x
+			var/oldPY = pixel_y
+			pixel_x = oldPY
+			pixel_y = (oldPX*(-1))
+
+
+
+//this is the main proc. It instantly moves our mobile port to stationary port S1
+//it handles all the generic behaviour, such as sanity checks, closing doors on the shuttle, stunning mobs, etc
+/obj/docking_port/mobile/proc/dock(obj/docking_port/stationary/S1)
+	. = canDock(S1)
+	if(.)
+		throw EXCEPTION("dock(): shuttle cannot dock")
+		return .
+
+	if(canMove())
+		return -1
+
+//		//rotate transit docking ports, so we don't need zillions of variants
+//		if(istype(S1, /obj/docking_port/stationary/transit))
+//			S1.dir = turn(NORTH, -travelDir)
+
+	var/obj/docking_port/stationary/S0 = get_docked()
+	var/turf_type = /turf/space
+	var/area_type = /area/space
+	if(S0)
+		if(S0.turf_type)
+			turf_type = S0.turf_type
+		if(S0.area_type)
+			area_type = S0.area_type
+
+	//close and lock the dock's airlocks
+	closePortDoors(S0)
```

Comment:
I guess.

Basically, -tg- already has it setup so that, when a shuttle departs, it will search for the closest airlock to the docking port controller (more complicated than that, but it basically closes the airlock next to the docking controller).
I modified it to work based on id_tags- Any airlocks with the same ID as a stationary dock (ex, emergency_home) will be closed, and, as an addition, they will also be locked.
Any airlock with the ID "s_docking_airlock" *aboard* the shuttle will also be bolted.